sinks/sink-common/src/error.rs

```diff
@@ -1,52 +1,161 @@
 use std::{fmt, process::ExitCode};
 
-use error_stack::Result;
+use error_stack::{report, Context, Report, Result, ResultExt};
 
 /// Sink error.
 ///
 /// Sink developers should default to returning `SinkError::Temporary` for all errors.
 /// `SinkError::Configuration` should be returned for configuration-related errors.
 /// `SinkError::Fatal` should only be returned from `Sink::handle_data` and `Sink::handle_invalidate`.
 #[derive(Debug)]
-pub enum SinkConnectorError {
+pub enum SinkError {
     /// Configuration error. Should not retry.
     Configuration,
     /// Temporary error. Should retry.
     Temporary,
     /// Fatal error. Should not retry.
     Fatal,
+    // Status server error
+    Status,
+    // Persistence error
+    Persistence,
+    // Load script error
+    LoadScript,
 }
 
 pub trait ReportExt {
     fn to_exit_code(&self) -> ExitCode;
 }
 
-impl error_stack::Context for SinkConnectorError {}
+impl error_stack::Context for SinkError {}
 
-impl fmt::Display for SinkConnectorError {
+impl fmt::Display for SinkError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            SinkConnectorError::Configuration => f.write_str("sink configuration error"),
-            SinkConnectorError::Temporary => f.write_str("temporary sink error"),
-            SinkConnectorError::Fatal => f.write_str("fatal sink error"),
-        }
+        // match self {
+        //     SinkError::Configuration => f.write_str("sink configuration error"),
+        //     SinkError::Temporary => f.write_str("temporary sink error"),
+        //     SinkError::Fatal => f.write_str("fatal sink error"),
+        //     _ => f.write_str("sink error"),
+        // }
+        f.write_str("sink error")
     }
```

Comment:
I think here you want to customize the error message based on the enum variant?

Reply:
the customized error message will come from the variant itself, example: https://github.com/apibara/dna/pull/300/files#diff-7b3eae91ded99e285af550c0e29ba30a4df9251721447596f0e8fd4e76641facR66